### DISCUSSION TITLE
Whats the best way to store and act with growing array

### DISCUSSION THREAD

**Discussion by ivan-suhorukov** [ai/nanostores#304] [Q&A] [answered]
Hello! Sorry for my stupid question, maybe. I want to store the chat messages. And it is an array of messages. What is the best way to store it? In atom or map or deepMap maybe? And i want to add messages in store while user types them, so whats the best way to do it? `setKey` or `const messages = store.get(); messages.push('hello'); store.set(messages)` maybe? Thanks!

**Comment by ai** [comment, marked as answer]
I suggest have `atom()` with chat history and separated atom with typing message `draft`

**Reply by ivan-suhorukov** [reply]
Ok, thanks, i will use it.

**Reply by muhajirdev** [reply]
something like const messages = atom<ChatMessage[]> ? 
Will it cause rerender of the whole messages, everytime I push new message to the atom?
